### PR TITLE
readme: add description of ECMAScript

### PR DESCRIPTION
**Pull request by unikounio**
Added a description of ECMEScript to the readme and adjust the sample files to match.
### Issue
[Docs: Add sample code ES Modules format to the examples in the README\. · Issue \#61 · minimistjs/minimist](https://github.com/minimistjs/minimist/issues/61)

### CODE REVIEW

File: README.md
Comment (ljharb) [inline README.md]:
```suggestion
// for CJS
const argv = require('minimist')(process.argv.slice(2));

// for ESM
// import minimist from 'minimist'; // (for ECMAScript)
// const argv = minimist(process.argv.slice(2));
```

to showcase the differences here

File: README.md
Comment (ljharb) [inline README.md]:
```suggestion
const parseArgs = require('minimist');
```

File: example/parse.js
Comment (ljharb) [inline example/parse.js]: let's revert this, so the one-liner is in the example.

i also don't think we need the git clone version; if it's a concrete issue then we can add a package.json or something in the example folder

**Review by ljharb — CHANGES_REQUESTED** [review]

File: README.md
Comment (shadowspawn) [reply README.md]: Short version: ok, yes

Long version: I personally dislike putting code on the require line, but it was the previous usage and is a pattern that was popular. Yargs used to show this too. Ok.

Comment (ljharb) [reply README.md]: it's a reason CJS is better than ESM :-)

File: example/parse.js
Comment (shadowspawn) [reply example/parse.js]: Short version: Ok.

Long version: I run the examples from repos as a library maintainer to check the examples work, and check functionality. It might be little used by end users, especially since they files are so small. Using a "normal" import does make the example cleaner and match end-user code, so probably a win.

Comment (ljharb) [reply example/parse.js]: it's totally fine to have a command that maintainers can run to test examples; the way to do that is to symlink the package into node_modules rather than altering the examples.

Comment (shadowspawn) [reply example/parse.js]: Oh, interesting. Perhaps out of scope for Minimist but I am interested for Commander. Light question, do you have an example of that pattern somewhere in your repos?

Comment (ljharb) [reply example/parse.js]: nah, i don't usually bother with it - but you can add `"minimist": "file:./"` to devDeps and it should work in most npm versions.

Comment (shadowspawn) [reply example/parse.js]: I was experimenting with what is needed to import from `'minimist'`, and using `var argv = require('minimist')(process.argv.slice(2));` in the example file and running it directly from the repo seems to work fine. Finds the parent package by name just fine.  Can apparently use the "normal" include without any extra setup!

Comment (ljharb) [reply example/parse.js]: If the package has `exports`, that will be true in node versions that support the `exports` field - since this package doesn't, i'm not sure why it would work :-/

**Comment by unikounio** [comment]
@ljharb @shadowspawn 
Thank you both for your thorough reviews. I have made the necessary updates based on your feedback.

File: example/parse.js
Comment (shadowspawn) [reply example/parse.js]: Urk. What on earth was I testing yesterday!? I will check again. (I have some theories...)

Comment (shadowspawn) [reply example/parse.js]: Ok, I found this a bit amusing, despite my error. I was fooled by `node_modules` including a hoisted `minimist` due to a transient dependency.

```
% node --version
v20.11.1
% npm install
...
% ls -ld node_modules/minimist
drwxr-xr-x  12 john  staff  384 13 Mar 21:31 node_modules/minimist
% n 0.12.18
     copying : node/0.12.18
   installed : v0.12.18 (with npm 2.15.11)
% node --version
v0.12.18
% cat example/parse.js
'use strict';

var argv = require('minimist')(process.argv.slice(2));
console.log(argv);
% node example/parse.js
{ _: [] }
```

**Review by shadowspawn — APPROVED** [review]
Looks good to me

**Comment by shadowspawn** [comment]
Thanks for working through our various comments @unikounio

File: README.md
Comment (ljharb) [inline README.md]:
```suggestion
// import minimist from 'minimist';
```

File: README.md
Comment (ljharb) [inline README.md:56]: https://github.com/minimistjs/minimist#var-argv--parseargsargs-opts needs to keep working forever, so:
```suggestion
<a name="var-argv--parseargsargs-opts"></a>
## const argv = parseArgs(args, opts={})
```

File: README.md
Comment (ljharb) [inline README.md]:
```suggestion
Example files: [example/parse.js](./example/parse.js) (CJS) / [example/parse.mjs](./example/parse.mjs) (ESM)
```
(since "CommonJS" is actually a bit broader than CJS, and "ECMAScript" is much broader than ESM)

**Review by ljharb — CHANGES_REQUESTED** [review]
Almost there :-)

**Comment by unikounio** [comment]
@ljharb
Thank you for the thoughtful suggestion. I have addressed it.

**Comment by shadowspawn** [comment]
Urk. Failing a couple of lint tests due to using "new" features in the example code (`const` and modules). To be clear, these would be problems if they were in the actual implementation since we support very old node versions.

My first thought is examples using modern syntax are more appropriate and useful for new users. Is it appropriate to add an override to `.eslintrc` for the examples folder and allow modern features?

Do you already have a preferred way of approaching this @ljharb ?

**Comment by ljharb** [comment]
yep! add an "overrides" object for `example/**` in the root eslintrc, and extend `@ljharb/eslint-config/node/latest`.

**Comment by shadowspawn** [comment]
Are you happy to look at this @unikounio , or would you like a hand with the eslint change? (Fairly easy hopefully, but depending on what you already know and what you are interested in learning...)

**Comment by unikounio** [comment]
@shadowspawn 
Thank you for your suggestion. I would like to do this ESLint configuration change. Would it be better to include this change in this PR or would it be more appropriate to create a new PR for this  change?

**Comment by shadowspawn** [comment]
Include in this PR thanks.

**Comment by unikounio** [comment]
Just pushed the changes to apply the latest ESLint config to the `examples` directory.
Please take a look when you have a moment. Thanks!

**Review by ljharb — APPROVED** [review]
rebased and squashed to one commit, and fixed formatting (and examples → example in eslintrc)

this LGTM and can be ff-merged once @shadowspawn concurs :-)

**Review by shadowspawn — APPROVED** [review]
LGTM 👍

**Comment by unikounio** [comment]
Thank you both so much for helping me with this PR!
@ljharb and @shadowspawn, I deeply appreciate your support and the valuable feedback you've provided.
I've learned a lot from this experience and am looking forward to any future opportunities to work together again!

**Review by Queenkaz — COMMENTED** [review]
👏

**Review by Queenkaz — COMMENTED** [review]
🤟🏽

**Pull request merged**
